prov/util/src/util_domain.c

```diff
@@ -112,8 +112,8 @@ int ofi_domain_init(struct fid_fabric *fabric_fid, const struct fi_info *info,
 	ret = ofi_mr_map_init(domain->prov, info->domain_attr->mr_mode,
 			      &domain->mr_map);
 	if (ret) {
-		ofi_domain_close(domain);
-		return ret;
+		ret_err = ofi_domain_close(domain);
+		return ret_err ? ret_err : ret;
```

Comment:
We should return the first error here.  We can probably just assert that domain_close succeeds.  There's no reason for it to fail in this case.  If coverity still complains, I can discard the warning.